src/test/compile-fail/extern-crate-visibility.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+mod foo {
+    extern crate core;
+    pub use self::core as reexported_core; // Check that private extern crates can be reexported
```

Comment:
The `private_in_public` warning is reported during lint pass, much later than resolve.
It looks like it's tested separately in `private-variant-and-crate-reexport.rs`.